File: app/src/main/java/com/yanxing/ui/animation/ObjectAnimationActivity.java
package com.yanxing.ui.animation;

import android.animation.ObjectAnimator;
import android.animation.ValueAnimator;
import android.support.design.widget.Snackbar;
import android.view.View;
import android.widget.ImageView;

import com.yanxing.base.BaseActivity;
import com.yanxing.model.Point;
import com.yanxing.ui.R;

import butterknife.BindView;
import butterknife.OnClick;

/**
 * 属性动画
 * Created by lishuangxiang on 2016/8/1.
 */
public class ObjectAnimationActivity extends BaseActivity {

    @BindView(R.id.image)
    ImageView mImage;

    @BindView(R.id.image1)
    ImageView mImage1;

    @Override
    protected int getLayoutResID() {
        return R.layout.activity_object_animation;
    }

    @Override
    protected void afterInstanceView() {

    }

    @OnClick({R.id.image,R.id.image1})
    public void onClick(View view) {
        switch (view.getId()){
            case R.id.image:
                ObjectAnimator
                .ofFloat(mImage,"rotationX",0.0f,360f)
                .setDuration(1000)
                .start();
                break;
            case R.id.image1:
                ObjectAnimator objectAnimator=ObjectAnimator
                        .setDuration(1000);
                objectAnimator.start();
                objectAnimator.addUpdateListener(new ValueAnimator.AnimatorUpdateListener() {
                    @Override
                    public void onAnimationUpdate(ValueAnimator animation) {
                        float f= (float) animation.getAnimatedValue();
                        mImage1.setScaleX(f);
                        mImage1.setScaleY(f);
                        mImage1.setAlpha(f);
                    }
                });
                Snackbar.make(mImage1, "Click", Snackbar.LENGTH_SHORT).show();
                break;
        }
    }

    public void paoWuXian(){
        Point startPoint=new Point(0,0);
        Point endPoint=new Point(300,300);
        ValueAnimator valueAnimator=ValueAnimator.ofObject(new PointTypeEvaluator(),startPoint,endPoint);
        valueAnimator.setDuration(500);
        valueAnimator.start();
    }
}

File: app/src/main/java/com/yanxing/ui/animation/ObjectAnimationActivity.kt
package com.yanxing.ui.animation

import android.animation.ObjectAnimator
import android.animation.ValueAnimator
import com.google.android.material.snackbar.Snackbar

import com.yanxing.base.BaseActivity
import com.yanxing.model.Point
import com.yanxing.ui.R

import kotlinx.android.synthetic.main.activity_object_animation.*

/**
 * 属性动画
 * Created by lishuangxiang on 2016/8/1.
 */
class ObjectAnimationActivity : BaseActivity() {


    override fun getLayoutResID(): Int {
        return R.layout.activity_object_animation
    }

    override fun afterInstanceView() {
        image.setOnClickListener {
            ObjectAnimator
                    .ofFloat(image, "rotationX", 0.0f, 360f)
                    .setDuration(1000)
                    .start()
        }
        image1.setOnClickListener {
            val objectAnimator = ObjectAnimator
                    .ofFloat(image1, "rotationX", 0.0f, 1.0f)
                    .setDuration(1000)
            objectAnimator.start()
            objectAnimator.addUpdateListener { animation ->
                val f = animation.animatedValue as Float
                image1.scaleX = f
                image1.scaleY = f
                image1.alpha = f
            }
            Snackbar.make(image1, "Click", Snackbar.LENGTH_SHORT).show()
        }
    }

    fun paoWuXian() {
        val startPoint = Point(0f, 0f)
        val endPoint = Point(300f, 300f)
        val valueAnimator = ValueAnimator.ofObject(PointTypeEvaluator(), startPoint, endPoint)
        valueAnimator.duration = 500
        valueAnimator.start()
    }
}
